lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -38,7 +38,9 @@ def self.run(params)
         if app_id.nil?
           UI.crash!(ErrorMessage::MISSING_APP_ID)
         end
-        upload(app_id, binary_path)
+        # Should be fine since we will have an error message if it goes over the limit of tries and every other time that it does not succed should end in some error so there shouldn't be any invalid input at this stage
+        # Release notes may need to check before passing it in that it isnt nil.
+        post_notes(app_id, upload(app_id, binary_path), params[:release_notes])
```

Comment:
I put this as a comment, but upload will only return if it succeeds so we should be ok in terms of any invalid input here.

Reply:
Pretty clever to inline the call to `upload`! In this case I would put it on its own line though, because the return value is unclear. I'd rather see: `release_id = upload(...` above so that I know we're passing a release ID to post_notes.

For example if the method was called `get_release_id` I think this would be fine. But that wouldn't make sense because `upload` is a much better description of what it does.

Reply:
This works out since I will need the release ID later for enable access.